moveit_ros/moveit_servo/src/pose_tracking.cpp

```diff
@@ -79,12 +79,9 @@ PoseTracking::PoseTracking(const ros::NodeHandle& nh,
 PoseTrackingStatusCode PoseTracking::moveToPose(const Eigen::Vector3d& positional_tolerance,
                                                 const double angular_tolerance, const double target_pose_timeout)
 {
-  // Roll back the target pose timestamp to ensure we wait for a new target pose message
-  target_pose_.header.stamp = ros::Time::now() - ros::Duration(2 * target_pose_timeout);
```

Comment:
I tried to explain on Slack that this line accounts for the use case where multiple waypoints are tracked, one after another. If a fresh target pose is late to arrive, we could begin moving toward the previous waypoint until a new target pose finally arrives (network latency). I saw this happen on a different project.

How about adding a public member function called `resetTargetPose()` where `target_pose_.header.stamp` is set to `ros::Time::now() - ros::Duration(2 * target_pose_timeout)` and all other fields are set to empty. That would give an entirely optional way to reset the target pose between waypoints.

If anybody has a better way to indicate that a std_msgs::Time field hasn't been initialized, that would be great. Here's the header file:  http://docs.ros.org/en/electric/api/std_msgs/html/Time_8h_source.html

Reply:
We could use a separate Boolean flag, too

Reply:
My thinking here was that since the function claims to servo the arm to a pose that is fresh enough, it should do only that. The target pose is already updated asynchronously by using callback function, so doesn't the caller side already have everything it needs? Going through waypoints can be achieved by calling `moveToPose` once and then after that again with a new target.

Reply:
`Going through waypoints can be achieved by calling moveToPose once and then after that again with a new target.`

The problem is, how do you know when the target pose is new? Will make a PR against your branch

Reply:
PRs are welcomed! And I think I finally understood what the problem is. You don't know whether the `target_pose_` has been updated prior to `moveToPose` function call. Would it make sense to remove the internal state altogether and just pass the latest pose as a parameter into the function? Or resetting as you proposed and waiting for a new pose inside the while loop.